### PR TITLE
Fixed the topic_entity_tag create_tag end point to check for existing tag and return the tag data (curie+name)

### PR DESCRIPTION
**Pull request by sweng66**

### CODE REVIEW

File: agr_literature_service/api/crud/topic_entity_tag_crud.py [hunk 1]
```diff
@@ -4,13 +4,15 @@
 """
 from collections import defaultdict
 from typing import Dict
```
Comment (azurebrd) [inline agr_literature_service/api/crud/topic_entity_tag_crud.py:6]: extra line ?

File: agr_literature_service/api/crud/topic_entity_tag_crud.py [hunk 2]
```diff
@@ -31,7 +33,7 @@
 ATP_ID_SOURCE_CURATION_TOOLS = "curation_tools"
 
 
-def create_tag(db: Session, topic_entity_tag: TopicEntityTagSchemaPost) -> int:
+def create_tag(db: Session, topic_entity_tag: TopicEntityTagSchemaPost) -> dict:
```
Comment (azurebrd) [inline agr_literature_service/api/crud/topic_entity_tag_crud.py:36]: oh, how did this work as an int before ?

Comment (azurebrd) [reply agr_literature_service/api/crud/topic_entity_tag_crud.py:36]: Oh, it used to be an int, nevermind

**Review by azurebrd — APPROVED** [review]
Great, thanks @sweng66 !  (feel free ignore that single comment, I wrote because I wrote what I thought was a useful different single comment, but it wasn't useful)

File: agr_literature_service/api/crud/topic_entity_tag_crud.py (hunk 2 above)
Comment (sweng66) [reply agr_literature_service/api/crud/topic_entity_tag_crud.py:36]: It was returning the newly generated topic_entity_tag_id before so it was int @azurebrd

Comment (sweng66) [reply agr_literature_service/api/crud/topic_entity_tag_crud.py:36]: I didn't pay attention to that 'int' earlier until the unittest failed =)

Comment (azurebrd) [reply agr_literature_service/api/crud/topic_entity_tag_crud.py:36]: Thanks @sweng66 :)

**Comment by sweng66** [comment]
Thanks @azurebrd

**Pull request merged**